packages/opentelemetry-instrumentation-http/src/http.ts

```diff
@@ -639,4 +652,20 @@ export class HttpInstrumentation extends InstrumentationBase<Http> {
       true
     );
   }
+
+  private _callStartIncomingSpanHook(request: http.IncomingMessage): SpanAttributes {
+    return safeExecuteInTheMiddle(
+      () => this._getConfig().startIncomingSpanHook!(request),
```

Comment:
You are already checking that ```startOutgoingSpanHook``` was defined before calling this function, so this seems redundant and  causes a lot of additional code that is not necessary (resulting in larger payload than necessary).

What about just
544: ```let hookAttributes = instrumentation._callStartOutgoingSpanHook(optionsParsed);```
```
private _callStartIncomingSpanHook(request: http.IncomingMessage): SpanAttributes {
   let hookFunc = this._getConfig().startIncomingSpanHook;
   if (typeof hookFunc === "function") {
        return safeExecuteInTheMiddle(
              () => hookFunc(request),
              () => { },
              true
        );
    }
    return undefined; // Not strictly required
}
 
```

Reply:
Thanks for advice, I'v removed redundant code!